pkg/print/list/base.go

```diff
@@ -172,16 +200,36 @@ func (b *BaseListPrinter) Print(ch <-chan event.Event, previewStrategy common.Dr
 			if err := formatter.FormatDeleteEvent(e.DeleteEvent); err != nil {
 				return err
 			}
+		case event.WaitType:
+			switch e.WaitEvent.Operation {
+			case event.Reconciled:
+				waitStats.incReconciled()
+			case event.ReconcileSkipped:
+				waitStats.incSkipped()
+			}
+			if e.DeleteEvent.Error != nil {
+				waitStats.incFailed()
+			}
+			if err := formatter.FormatWaitEvent(e.WaitEvent); err != nil {
+				return err
+			}
 		case event.ActionGroupType:
-			if err := formatter.FormatActionGroupEvent(e.ActionGroupEvent, actionGroups, applyStats,
-				pruneStats, deleteStats, statusCollector); err != nil {
+			if err := formatter.FormatActionGroupEvent(
+				e.ActionGroupEvent,
+				actionGroups,
+				applyStats,
+				pruneStats,
+				deleteStats,
+				waitStats,
+				statusCollector,
+			); err != nil {
 				return err
 			}
 		}
 	}
 	failedSum := applyStats.Failed + pruneStats.Failed + deleteStats.Failed
-	if failedSum > 0 {
-		return fmt.Errorf("%d resources failed", failedSum)
+	if failedSum > 0 || waitStats.Failed > 0 {
```

Comment:
TODO: This needs to be update for the new WaitEvent Operations. There's no more Failed/Error. Do we want to return Timeouts in the error?

Reply:
hows it look now?